dism resource: add sourcePath parameter to allow for offline feature installs (#1561)
PowerShell/DSC: docs/reference/resources/Microsoft/Windows/FeatureOnDemandList/index.md

  - [capabilities](#capabilities) - An array of capability entries.

> - **Instance properties:** <a id="instance-properties"></a> The following properties are optional.
>   They define the desired state for an instance of the resource.
> 
>   - [sourcePaths](#sourcePaths) - The location of the source files to use for installation if 
>     necessary.

Reviewer: I think this is more correctly a write-only property, given that it only affects how the resource behaves during a **Set** operation.

Instance properties typically represent actual properties of the underlying resource (like the name of the feature and whether it exists) where write-only properties affect how the resource behaves and read-only properties are metadata about the resource that the user can't directly and idempotently manage (like the last write time for a file or the author for a software package).

```suggestion
- **Write-only properties:** <a id="write-only-properties"></a> The resource accepts the following
  properties that affect its behavior but can't be retrieved for the actual state of an instance.
  For more information about read-only properties, see the "Write-only resource properties" section
  in [DSC resource properties][aa].

  - [sourcePaths](#sourcePaths) - The location of the source files to use for installation if 
    necessary.
```

The size in bytes that the capability occupies on disk after installation. This property is returned
by **Get** and **Export** operations.

### sourcePaths

> <details><summary>Expand for <code>sourcePaths</code> property metadata</summary>
> 
> ```yaml
> Type       : array
> IsRequired : false
> IsKey      : false
> IsReadOnly : false
> ```
> 
> </details>
> 
> Supplied at the top level of the **Set** operation, indicates the location of the source files to
> use for installation if necessary. The DISM API will search these paths if the feature files are
> not available in the local feature store. See the [feature on demand repository documentation][08]
> for more information on valid sources.
> 
> This property is optional and will be omitted from the response if no value is provided.

Reviewer: ``````suggestion
<details><summary>Expand for <code>sourcePaths</code> property metadata</summary>

```yaml
Type              : array
ItemsType         : string
ItemsMinimumCount : 1
ItemsMustBeUnique : true
IsRequired        : false
IsKey             : false
IsReadOnly        : false
IsWriteOnly       : true
```

</details>

Indicates where the resource should look for source files to install a feature when the feature
files aren't available in the local feature store for offline systems. Specify this property as an
array of one or more unique strings where each string represents the path to a source file to use.

This property affects how the resource behaves during **Set** operations. It has no affect for any
other resource operations. For more information about valid sources for the DISM API, see the
[`/Enable-Feature` reference for the `dism` command][08].
``````

        "additionalProperties": true
      }
    },

>     "sourcePaths": {
>       "type": "array",
>       "items": {
>         "type": "string"
>       }
>     },

Reviewer: ```suggestion
    "sourcePaths": {
      "type": "array",
      "items": {
        "type": "string"
      },
      "minItems": 1,
      "uniqueItems": true
    },
```

[05]: ../../../../../concepts/resources/properties.md#read-only-resource-properties
[06]: ../OptionalFeatureList/index.md
[07]: /windows-hardware/manufacture/desktop/features-on-demand-v2--capabilities

> [08]: /windows-hardware/manufacture/desktop/features-on-demand-v2--capabilities?view=windows-11#fod-repositories

Reviewer: ```suggestion
[08]: /windows-hardware/manufacture/desktop/features-on-demand-v2--capabilities?view=windows-11#fod-repositories
[aa]: ../../../../../concepts/resources/properties.md#write-only-resource-properties
```
